functions/vcs.zsh

```diff
@@ -110,7 +110,18 @@ function +vi-hg-bookmarks() {
 
 function +vi-vcs-detect-changes() {
   if [[ "${hook_com[vcs]}" == "git" ]]; then
-    vcs_visual_identifier='VCS_GIT_ICON'
+
+    local remote=`git ls-remote --get-url`
+    if [[ "$remote" =~ "github" ]] then
```

Comment:
@davidmpaz Returning values in ZSH is quite a pain. There are couple of ways to accomplish that (worst first):

1. Use an cache file
    Obviously this is quite some overkill for just one value, but sometimes this is the way to go.
2. Control it via return codes
    You could return an integer as return code in your function and have that interpreted on the caller side of the function. This is stupid, if you return more than `true` (0) or `false` (1) and the return values always confuse me.
3. Fill a global variable
    In the callee function, fill a global variable. This is often done with handling associative arrays.
4. Echo the value
    If you echo the value you want in the callee function, you could catch it by calling it in a subshell (`my_value=$(call_me)`).

For sure I missed some more or less worse method. But to stay on topic, you could do something like (pseudocode):
```zsh
function buildVcsVisualIdentifier() {
    local remote=$(git ls-remote --get-url 2> /dev/null)
    if [[ "$remote" =~ "github" ]] then
        vcs_visual_identifier='VCS_GIT_GITHUB_ICON'
    elif [[ "$remote" =~ "bitbucket" ]] then
        vcs_visual_identifier='VCS_GIT_BITBUCKET_ICON'
    elif [[ "$remote" =~ "gitlab" ]] then
        vcs_visual_identifier='VCS_GIT_GITLAB_ICON'
    else
        vcs_visual_identifier='VCS_GIT_ICON'
    fi

    elif [[ "${hook_com[vcs]}" == "hg" ]]; then
        vcs_visual_identifier='VCS_HG_ICON'
    elif [[ "${hook_com[vcs]}" == "svn" ]]; then
        vcs_visual_identifier='VCS_SVN_ICON'
    fi
}

function +vi-vcs-detect-changes() {
    buildVcsVisualIdentifier()
    ...
}
```

The `vcs` segment is a quite complex one. It interacts heavily with ZSHs own VCS_INFO subsystem. In that subsystem we register a couple of git-hooks (like `vi-vcs-detect-changes`). So these hooks get called by VCS_INFO before the `vcs_prompt` (segment main-) function gets called. Long story short: We cannot control the caller side of the function, that is why we set the `vcs_visual_identifier` as a global variable (which is read in `vcs_prompt`).